src/audio/volume/volume.c

```diff
@@ -121,6 +126,9 @@ static enum task_state vol_work(void *data)
 		vol_sync_host(cd, i);
 	}
 
+	if (!again)
+		cd->vol_ramp_active = 0;
+
 	/* do we need to continue ramping */
 	return again ? SOF_TASK_STATE_RESCHEDULE : SOF_TASK_STATE_COMPLETED;
```

Comment:
nitpick: with your addition I'd regroup this as
```
if (again)
	return SOF_TASK_STATE_RESCHEDULE;

cd->vol_ramp_active = false;

return SOF_TASK_STATE_COMPLETED;
```

Reply:
Yep, makes sense. I was about to do this myself as well.

Reply:
done